### PR TITLE
DRILL-6217: NaN/Inf NestedLoopJoin processes NaN values incorrectly

### PR DESCRIPTION
**Pull request by vladimirtkach**
- Changed loggic for equality functions to handle NaN values as the biggest ones

### CODE REVIEW

**Comment by arina-ielchiieva** [comment]
Please add unit tests for each join type (good example - TestJoinEmptyDirTable.java)

**Comment by vrozov** [comment]
@vladimirtkach Use JIRA in PR title (to automatically update JIRA with a link to the PR and PR updates).

File: exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestJsonNanInf.java [hunk 1]
```diff
@@ -331,4 +343,63 @@ public void testInnerJoinWithNaN() throws Exception {
     }
   }
 
+  @Test
+  public void testHashJoinWithNaN() throws Exception {
```
Comment (arina-ielchiieva) [inline exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestJsonNanInf.java]: When do you reset join options (`JoinTestBase.resetJoinOptions()`)?

Comment (vladimirtkach) [reply exec/java-exec/src/test/java/org/apache/drill/exec/vector/complex/writer/TestJsonNanInf.java]: @arina-ielchiieva added resetting of default options

**Comment by arina-ielchiieva** [comment]
+1, thanks for making the changes.

**Pull request closed without merging**